autogpt_platform/backend/backend/data/rabbitmq.py

```diff
@@ -0,0 +1,280 @@
+from enum import Enum
+import logging
+from typing import Optional, Callable, Any
+from abc import ABC, abstractmethod
+
+import pika
+from pika.spec import BasicProperties
+import aio_pika
+from backend.util.retry import conn_retry
+from backend.util.settings import Settings
+from pydantic import BaseModel
+
+logger = logging.getLogger(__name__)
+
+
+class ExchangeType(str, Enum):
+    DIRECT = "direct"
+    FANOUT = "fanout"
+    TOPIC = "topic"
+    HEADERS = "headers"
+
+
+class Exchange(BaseModel):
+    name: str
+    type: ExchangeType
+    durable: bool = True
+    auto_delete: bool = False
+
+
+class Queue(BaseModel):
+    name: str
+    durable: bool = True
+    auto_delete: bool = False
+    # Optional exchange binding configuration
+    exchange: Optional[Exchange] = None
+    routing_key: Optional[str] = None
+    arguments: Optional[dict] = None
+
+
+class RabbitMQConfig(BaseModel):
+    """Configuration for a RabbitMQ service instance"""
+
+    vhost: str = "/"
+    exchanges: list[Exchange]
+    queues: list[Queue]
```

Comment:
Expect this to change a bit as I get used to working with it. I'm thinking de-duplicating exchanges and queues is probably going to happen